packages/grid/x-data-grid/src/components/base/GridOverlays.tsx

```diff
@@ -29,24 +48,23 @@ function GridOverlayWrapper(props: React.PropsWithChildren<{}>) {
 
   let height: React.CSSProperties['height'] = viewportInnerSize?.height ?? 0;
   if (rootProps.autoHeight && height === 0) {
-    height = 'auto';
+    height = getMinimalContentHeight(apiRef); // Give room to show the overlay when there no rows.
   }
 
   if (!viewportInnerSize) {
     return null;
   }
 
   return (
-    <div
-      style={{
-        height,
-        width: viewportInnerSize?.width ?? 0,
-        position: 'absolute',
-        top: totalHeaderHeight,
-        bottom: height === 'auto' ? 0 : undefined,
-      }}
-      {...props}
-    />
+    <GridOverlayWrapperRoot>
+      <GridOverlayWrapperInner
```

Comment:
You also need to set the class names.

![image](https://user-images.githubusercontent.com/42154031/192060251-99e056ed-980f-482c-a3e7-a24dcd9c0875.png)